### PR TITLE
[Ready for Audit] Create ERC721 Lockable Module

### PR DESCRIPTION
**Pull request by darkty0x**

### CODE REVIEW

File: erc721-lockable/contracts/base/ERC721Lockable.sol
Comment (deep-quality-dev) [inline erc721-lockable/contracts/base/ERC721Lockable.sol:304]: This will check if this `tokenId` was already locked before burning, correct?

Comment (kazunetakeda25) [reply erc721-lockable/contracts/base/ERC721Lockable.sol:304]: Yes, correct @deep-quality-dev

**Review by deep-quality-dev — COMMENTED** [review]
Overall, Looks good to me, do we check lock status before approval?
- isApprovedForAll
- setApprovalForAll
- _beforeApprove

**Comment by kazunetakeda25** [comment]
No, we don't. They will be just ERC721 methods and they work as usual.
But instead, we do hae isLockAPprovedForAll method, and they will check lock status before approval.

**Comment by deep-quality-dev** [comment]
> > Overall, Looks good to me, do we check lock status before approval?
> > 
> > * isApprovedForAll
> > * setApprovalForAll
> > * _beforeApprove
> 
> No, we don't. They will be just ERC721 methods and they work as usual. But instead, we do hae isLockAPprovedForAll method, and they will check lock status before approval.

That's fine, it's up to requirements, in some cases, it requires to check lock status while approval. Your current lockable contract is much flexible then. Great

**Review by deep-quality-dev — APPROVED** [review]

File: erc721-lockable/contracts/base/ERC721Lockable.sol
Comment (solcmoon) [inline erc721-lockable/contracts/base/ERC721Lockable.sol:22]: If this is main contract, please remove `abstract`

Comment (solcmoon) [reply erc721-lockable/contracts/base/ERC721Lockable.sol:22]: and move this contract into root path

File: erc721-lockable/contracts/base/ERC721Lockable.sol
Comment (solcmoon) [inline erc721-lockable/contracts/base/ERC721Lockable.sol:38]: if you don't need this function, please remove this function and `IBunzz` interface

**Pull request closed without merging**
